meshtastic/interfaces/ble/lifecycle_primitives.py

```
@@ -0,0 +1,679 @@
+                )
+        set_event = getattr(coordinator, "set_event", None)
+        if callable(set_event) and not _is_unconfigured_mock_callable(set_event):
+            for event_name in event_names:
+                try:
+                    set_event(event_name)
+                except Exception:  # noqa: BLE001 - non-critical wake stays best effort
+                    logger.debug(
+                        "Error in thread_coordinator.set_event fallback for %s",
+                        event_name,
+                        exc_info=True,
+                    )
```

Ruff (0.15.6)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)